src/core/inc/Serializer.h

```diff
@@ -22,15 +31,10 @@ class Serializer {
     Serializer& begin_block(uint32_t type);
     Serializer& end_block();
 
-    Serializer& write(int8_t value);
-    Serializer& write(int16_t value);
-    Serializer& write(int32_t value);
-    Serializer& write(int64_t value);
-    Serializer& write(uint8_t value);
-    Serializer& write(uint16_t value);
-    Serializer& write(uint32_t value);
-    Serializer& write(uint64_t value);
-    Serializer& write(const std::string& s);
+    template <typename T>
+    Serializer& write(typename TypeTag<T>::type value) {
+        typename std::conditional<std::is_same<T, T>::value, void, char>::type sfinae[0];
```

Comment:
Jeśli nie podasz jawnie typu szablonu, to kompilator nie wybierze żadnej specjalizacji, i trafi tutaj. Ta linijka gwarantuje błąd kompilacji w takim wypadku (opierając się nieco o SFINAE).